evm/its.js

```diff
@@ -87,6 +87,24 @@ function compare(contractValue, configValue, variableName) {
     }
 }
 
+function isValidDestinationChain(config, chain, destinationChain, originalChain = null) {
+    const chains = config.chains;
+
+    const validDestination = Object.values(chains).some((chainObject) => chainObject.id === destinationChain);
+
+    if (!validDestination) {
+        throw new Error(`Invalid destination chain: ${destinationChain}`);
+    }
+
+    if (chain.id === destinationChain) {
+        throw new Error(`Destination chain ${destinationChain} equals source chain ${chain.id}`);
+    }
+
+    if (originalChain && destinationChain === originalChain) {
```

Comment:
Add these checks as well because I assume there would be no instance where the destination chain should equal the source chain or original chain (for factory functions)